core/mondoo-okta-security.mql.yaml

```diff
@@ -1,379 +1,1985 @@
-          ```
-      remediation: "### Use the Okta Admin Console\n\n1. Log into the organization as an administrator.\n2. Select **Security > Global Session Policy**.\n3. Select the rule to modify (select the pencil icon).\n4. Under **Session management**, select **Set time limit (Recommended)** and specify one day or lower.\n5. Under **Expire session after user has been idle on Okta for**, specify two hours or lower.\n6. Select **Update rule**.\n"
-  - uid: mondoo-okta-security-okta-auth-openid-saml
-    title: Use OpenID or SAML authentication for app access
-    impact: 80
+        Low
+
+        If a factor is required as part of the MFA enrollment policy, end users must enroll in the factor before they can sign in to their org. Setup varies depending on the factor specified.
+    variants:
+      - uid: mondoo-okta-security-okta-mfa-access-runtime
+  - uid: mondoo-okta-security-okta-mfa-access-runtime
+    title: Require at least one factor in every MFA enrollment policy
+    impact: 100
+    filters: 
+      asset.platform == "okta-org"
     mql: |
-      okta.applications.all(_.signOnMode == "OPENID_CONNECT" || _.signOnMode == "SAML_2_0")
+      everyoneGroupId = okta.groups.where( profile['name'] == "Everyone" ).map(id)[0]
+      okta.policies.mfaEnroll.one( status == "ACTIVE" && conditions['people']['groups']['include'].contains(everyoneGroupId) )
+    docs:
+      remediation: 
+        - id: console
+          desc: |
+            ### Set a required factor in an MFA enrollment policy
+            1. In the Admin Console, go to **Security > Multifactor**. The **Factor Types** page appears.
+            2. Click **Factor Enrollment** to switch to factor enrollment policies and rules.
+            3. Select a policy and click **Edit** to modify it.
+            4. From the list of eligible factors, set at least one factor to **Required**.
+            5. Click **Update Policy**.
+
+            ### Prompt an end user to enroll in a required factor
+            To prompt an end user to enroll in a required factor, you may do one of the following:
+
+            - Set an Okta sign-on policy rule that prompts a user for factor enrollment.
+            - Set a factor enrollment policy rule that allows a user to enroll in a factor when challenged for MFA.
+            - Set a factor enrollment policy rule that prompts the user to enroll in a factor the first time they sign in to their org.
+
+            ### Create an Okta sign-on policy rule that prompts for factor enrollment
+            1. From the Admin Console menu, click Security > Authentication. The Authentication policies page appears.
+            2. Click Sign On to access Sign-On Policies.
+            3. Select the policy and from the list of associated rules, click Edit to start modifying an existing policy rule. You can also create a rule.
+            4. From the Edit Rule window, select Prompt for Factor.
+            5. Click **Update Rule**.
```

Comment:
```suggestion
            5. Select **Update Rule**.
```

Reply:
@tas50 these are not our steps. they are from Okta's docs. They are copy/pasted.